docs/autogen.py

```diff
@@ -53,19 +53,15 @@ def get_function_signature(function, method=True):
         args = args[: -len(defaults)]
     else:
         kwargs = []
-    st = "%s.%s(" % (clean_module_name(function.__module__), function.__name__)
+    st = ["%s.%s(" % (clean_module_name(function.__module__), function.__name__)]
 
     for a in args:
-        st += str(a) + ", "
+        st.append(str(a))
     for a, v in kwargs:
         if isinstance(v, str):
-            v = "'" + v + "'"
-        st += str(a) + "=" + str(v) + ", "
-    if kwargs or args:
-        signature = st[:-2] + ")"
-    else:
-        signature = st + ")"
-    return signature
+            v = "'{}'".format(v)
+        st.append("{}={}".format(a, v))
+    st = ', '.join(st) + ")"
```

Comment:
you're not returning anything. If this didn't fail, then maybe we should have a test to make sure this should have failed

Reply:
complete 👍